src/StackExchange.Redis/RedisTransaction.cs

```diff
@@ -146,6 +154,33 @@ public QueuedMessage(Message message) : base(message.Db, message.Flags | Command
                 Wrapped = message;
             }
 
+            // for transactions, the inner operations should only be marked completed when the the final EXEC has been processed
+            /// <summary>
+            /// For queued messages (InnerOperations) in a the transaction, we cannot actually mark it complete until the exec
+            /// function has returned with the state of the transaction.
+            ///
+            /// Calling the base complete resets the ResultBox, 
+            /// </summary>
+            public override void Complete()
+            {
+                // still need to activate continuations for GetMessages(),
+                // which might be waiting for the last innerOperation to
+                // complete.
+                ResultBox?.ActivateContinuations();
```

Comment:
If I understood the logic behind this correct, in order for the last section of the GetMessage() logic to work, where it waits on the last result box of the innerOperations, this needs to be fired.

This ultimately means that in the lifetime of a QueuedMessage, it will fire/pulse twice. Oncee when it's marked Complete() after it completes it's call, and then again when TransactionComplete() is called when the TransactionMessage runs it's Complete(). (which calls TransactionComplete() on all its innerOperations).